platform/linux-generic/odp_hash_crc_gen.c

```diff
@@ -87,7 +87,7 @@ static inline uint32_t reflect_u32(uint32_t u32)
 	u8[2] = reflect_u8((u32 & 0x0000ff00u) >> 8);
 	u8[3] = reflect_u8(u32 & 0xffu);
 
-	return (u8[3] << 24) | (u8[2] << 16) | (u8[1] << 8) | u8[0];
+	return ((uint32_t)u8[3] << 24) | (u8[2] << 16) | (u8[1] << 8) | u8[0];
```

Comment:
This still makes the assumption that int is larger than 24 bits. Maybe just change the type of u8 from uint8_t to uint32_t (an added bonus: no casting needed). reflect_u24 has the same problem. Similarly, reflect_u16() assumes that int is larger than 16 bits. IOW, this change avoids undefined behaviour only with some additional non-obvious assumptions.